parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+                         r'<small>\[\[#[a-z_\-\' ]+\|\'\'followup\'\']]</small>',
+                         # Remove followup links in <small> tags
+                         r'<small>\'\'[a-z0-9 /]+\'\'</small>',  # Remove text in <small> tags
+                         r'<ref>.*?</ref>',  # Remove text in <ref> tags
+                         r'<nowiki>.*?</nowiki>',  # Remove text in <nowiki> tags
+                         ]
+    for regex in regexps_empty_sub:
+        text = re.sub(regex, '', text, flags=re.IGNORECASE)
+
+    regexps_sub_text = [r'\[\[([a-zé().:\',\- ]+)]]',  # Replace links such as [[Shitty Wizard]]
+                        r'\[\[[a-zé0-9().:\'/ ]+\|([a-zé().:\' ]+)]]',
+                        # Replace links such as  [[Ancient (Building)|Ancients]] and [[:File:Axe|Axe]]
+                        # r'{{h\|([a-zé().:\' ]+)}}',  # Replace hero names
+                        r'{{tooltip\|(.*?)\|.*?}}',  # Replace tooltips
+                        r'{{note\|([a-z.!\'\-?, ]+)\|[a-z.!\'\-?,()/ ]+}}',  # Replace notes
+                        ]
+    for regex in regexps_sub_text:
+        text = re.sub(regex, '\\1', text, flags=re.IGNORECASE)
+
+    if any(escape in text for escape in ['[[', ']]', '{{', '}}', '|', 'sm2']):
+        logger.warn('Response could not be processed : ' + text)
+
+    return text.strip()
+
+
+def clean_response_text(response_text):
+    """Method for preprocessing the given response text.
+    It:
+    * removes trailing and leading spaces
+    * removes all punctuations
+    * removes double spaces
+    * changes to lowercase
+
+    :param response_text: the response text to be cleaned
+    :return: cleaned response text
+    """
+    response_text = response_text.translate(str.maketrans(string.punctuation, ' ' * len(string.punctuation)))
+
+    while '  ' in response_text:
+        response_text = response_text.replace('  ', ' ')
+
+    response_text = response_text.strip().lower()
+
+    return response_text
+
+
+def links_for_files(files_list):
+    """Method that queries MediaWiki API used by Gamepedia to return links to the files list passed.
+    Does batch processing to avoid max number of files limit and header size limit.
+    Used asynchronous requests for faster processing.
+    Removes files version as we only need the latest one.
+
+    MediaWiki allows max 50 files(titles) at once : https://www.mediawiki.org/wiki/API:Query.
+
+    :param files_list: list of files
+    :return files_link_mapping: dict with file names and their links. dict['file'] = link
+    """
+
+    files_link_mapping = {}
+    futures = []
+    empty_api_length = len(requests.get(url=API_PATH, params=get_params_for_files_api([])).url)
+
+    with FuturesSession() as session:
+        files_batch_list = []
+        title_length = 0
+
+        for file in files_list:
+            # If header size overflows of number of files reaches the limit specified by MediaWiki, i.e. 50.
```

Comment:
"overflows of number" should probably be "overflows the number".

Reply:
Fixed in a514fffcb8cb409b8e1c18f991acbfb90fc59b3a